### PR TITLE
Use proper slf4j message patterns with exceptions

### LINKED ISSUE #441: SL4J handles exceptions specially which causes incorrect logs
SL4J special cases an exception in the case: `log.warn("{}", exception)`. This results in: `{}`

as SL4J generates special printing of the exception and so ignores it in substitution. This exists in various places, that I know of:

CassandraClientPoolingContainer: `log.warn("Not reusing resource {} due to {}", resource, e);`
CassandraVerifier: `log.error("Couldn't use host {} to create keyspace, it returned exception \"{}\" during the attempt.", host, f);`

### PR DESCRIPTION
**Pull request by schlosna**
Fixes #441

### CODE REVIEW

**Comment by schlosna** [comment]
Rebased to pickup fix for flaky test #454

@joelea for review / delegation.

**Comment by joelea** [comment]
Is there a reason behind the different choices between `getMessage` and `toString`?

Either way it looks reasonable.

**Comment by schlosna** [comment]
Main reason I prefer `Throwable#toString()` is that it includes exception class and localized message (or just message if not i18n), see http://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#toString--

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ protected int getPoolSize() {
             return f.apply(resource);
         } catch (Exception e) {
             if (isInvalidClientConnection(e)) {
-                log.warn("Not reusing resource {} due to {}", resource, e);
+                log.warn("Not reusing resource {} due to {}", resource, e.toString());
```
Comment (jboreiko) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: Are we happy with this no longer printing the stack?  toString (generally) will only return the exception name : Message of Exception as compared to the log method which will return the whole trace.

Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: Submitted PR #480 to address this.

**Pull request merged**